readthedocs/projects/models.py

```diff
@@ -144,11 +144,19 @@ class AddonsConfig(TimeStampedModel):
     Everything is enabled by default.
     """
 
-    DOC_DIFF_DEFAULT_ROOT_SELECTOR = "[role=main]"
-    LINKPREVIEWS_DEFAULT_ROOT_SELECTOR = "[role=main] a.internal"
-    LINKPREVIEWS_DOCTOOL_NAME_CHOICES = (
-        ("sphinx", "Sphinx"),
-        ("other", "Other"),
+    DEFAULT_ROOT_SELECTOR = ", ".join(
+        [
+            "main > div > div.md-content",  # Material for MkDocs
+            "article#main",  # Docsify
+            "article",  # Pelican, Docusaurus, Antora, Jekyll
+            "div#content",  # AsciiDoctor
+            "[role=main]",
+            "main",
+            "div.body",
+            "div.document",
```

Comment:
Related to what @stsewd pointed out, we might want to sort these all by specificity? I see some out of order.

Specificity is a value that can be calculated:
https://specificity.keegan.st/

We could also increase specificity for each selector to align ordering of specificity with how popular the tool is. For instance, the Sphinx selector can have higher specificity as `div.document[role=main]` and can then be the first in the list too. It's the most common tool and so would already be the most common selector match